### PR TITLE
Export StyleObject type from styletron-react package

### PR DESCRIPTION
**Pull request by chasestarr**
closes https://github.com/styletron/styletron/issues/341

### CODE REVIEW

**Comment by rtsao** [comment]
I think there might need to be some additional changes to the style types (for newer versions of Flow) so might want to hold off exporting them until that is more settled.

**Comment by phou87** [comment]
How are we looking on this?

**Comment by phou87** [comment]
Ping - given that this style is already exported, just in styletron and not styletron-react, any changes will need to be resolved anyways (anyone wishing to import the type currently will just do it from the former package), so I don't see the downside of exporting this now to save people a direct dependency?

**Comment by chasestarr** [comment]
Given that this PR improves type ergonomics and does not solve a bug nor add a feature we're going to hold off and make a single change once everything is ready to publish

**Comment by phou87** [comment]
Ok, is there an ETA for this?

File: packages/styletron-react/src/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
   driver,
   getInitialStyle,
   type StandardEngine,
-  type StyleObject,
+  type StyleObject as StdStyleObject,
```
Comment (rtsao) [inline packages/styletron-react/src/index.js]: Won't this break existing usages of `StyleObject`?
```suggestion
  type StyleObject,
```

File: packages/styletron-react/src/index.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ import {
 } from "./dev-tool.js";
 
 export {DebugEngine};
+export type StyleObject = StdStyleObject;
```
Comment (rtsao) [inline packages/styletron-react/src/index.js]:
```suggestion
export type {StyleObject};
```

File: packages/styletron-react/src/index.js (hunk 1 above)
Comment (rtsao) [reply packages/styletron-react/src/index.js]: Nevermind, type declarations are hoisted. But I still think we can just export the actual name

**Pull request merged**